### ISSUE TITLE
drush site-install and relative path for SQLite not working

### ISSUE THREAD

**Issue by alimac** [drush-ops/drush#1336]
Documentation (http://www.drushcommands.com/drush-7x/site-install/site-install) shows this example to use with SQLite:
```
drush site-install --db-url=sqlite://sites/example.com/files/.ht.sqlite
```
But when I use the above command, I get an error:
```
drush si -y --db-url=sqlite://sites/all/files/.ht.sqlite
You are about to CREATE the '/sites/all/files/.ht.sqlite' database. Do you want to continue? (y/n): y
Directory / exists, but is not writable. Please check directory permissions.                                                        [error]
SQLITE: Cannot create /sites/all/files   
```
^^ it looks like drush is trying to write to the root directory? 
When I use the full path:
```
drush si standard -y --db-url=sqlite://Users/alimac/drupal/sites/all/files/.ht.sqlite
```
installation works. 

Thing is, the relative path worked for me a couple days ago. I did `composer global update` and updated drush yesterday. Can anyone else reproduce this issue?

**Comment by weitzman** [comment]
The tests are failing with same error so something changed here - https://travis-ci.org/drush-ops/drush/builds/61097143

**Comment by weitzman** [comment]
I stepped through the code and this broke with 8027687a96fdebc0d71527cbc8702e3dad5bc0ea (Move uri cache from runserver to qd). Hopefully @greg-1-anderson  can take a look.

**Comment by jonhattan** [comment]
Fixed in #1589

**Comment by kenorb** [comment]
Confirming it works with the recent dev-master.
